code/game/objects/items/stacks/medical.dm

```diff
@@ -85,12 +95,21 @@
 		user.visible_message("<span class='notice'>[user] starts to apply [src] on [user.p_them()]self...</span>", "<span class='notice'>You begin applying [src] on yourself...</span>")
 		if(!do_mob(user, M, self_delay))
 			return
+		//After the do_mob to ensure metabolites have had time to process at least one tick. 
+		if(reagent && (C.reagents.get_reagent_amount(/datum/reagent/metabolite/medicine/styptic_powder) || C.reagents.get_reagent_amount(/datum/reagent/metabolite/medicine/silver_sulfadiazine)))
+			to_chat(user, "<span class='warning'>That stuff really hurt! You can't bring yourself to apply [src] to your wounds again yet, maybe someone else can help put it on for you.</span>")
+			return
 
 	user.visible_message("<span class='green'>[user] applies [src] on [M].</span>", "<span class='green'>You apply [src] on [M].</span>")
-
-	if(affecting.heal_damage(heal_brute, heal_burn) || stop_bleeding)
+	if(reagent)
+		reagents.reaction(M, PATCH, affecting = affecting)
+		M.reagents.add_reagent_list(reagent) //Stack size is reduced by one instead of actually removing reagents from the stack.
 		C.update_damage_overlays()
-		use(1)
+	use(1)
+
+/obj/item/stack/medical/on_grind()
+	reagents.clear_reagents() //By default grinding returns all contained reagents + grind_results, and for stackable items we only want grind_results
```

Comment:
yeah, but I think these should not have grind_results, and if you have manual reagent removal stuff, it's not necessary here.

Reply:
See above - it's a valid opinion and the code is slightly snowflakey to handle it this way, but the two alternatives were substantially worse in my opinion.